### PR TITLE
add dependabot and pr testing

### PR DESCRIPTION
**Pull request by ebarboni**
just trying to copy thinks from nbpacker. 
dependabot + Pr testing.

### CODE REVIEW

File: .github/workflows/test.yaml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  build:
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v3
+    - name: Set up JDK 11
+      uses: actions/setup-java@v3
+      with:
+        java-version: '8'
```
Comment (neilcsmith-net) [inline .github/workflows/test.yaml:36]: Name of step is still JDK 11.  Meant to be 8 or 11?

**Review by neilcsmith-net — APPROVED** [review]
Couple of minor comments.  Looks good.  Can't make my mind up whether I like dependabot!

File: .github/workflows/test.yaml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+        distribution: 'temurin'
+        cache: maven
+    - name: Build with Maven
+      run: mvn clean install
```
[the remark this answers is not in the document]
Comment (ebarboni) [reply .github/workflows/test.yaml]: I'm not good at copy paste :D

File: .github/workflows/test.yaml (hunk 1 above)
Comment (ebarboni) [reply .github/workflows/test.yaml:36]: will keep 8 for now. we have no use of java yet but not sure how it will works with jni

**Pull request merged**